micrometer-observation/src/main/java/io/micrometer/observation/Observation.java

```diff
@@ -146,21 +153,21 @@ static <T extends Context> Observation createNotStarted(@Nullable ObservationCon
      */
     static Observation start(ObservationConvention<? extends Context> observationConvention,
             @Nullable ObservationRegistry registry) {
-        return start(observationConvention, null, registry);
+        return start(observationConvention, () -> null, registry);
```

Comment:
I couldn't get rid of this `() -> null` lambda because of generics. I wonder if we need this method. What's the use case to provide an `ObservationConvention` that doesn't target `Context` and not provide a context? If it only ever targets `Context`, then we don't need the `? extends Context` and then we can change the lambda to `Context::new`

Reply:
I'll update the parameter type to `ObservationConvention<Context>` since having another Context type when a `Context` will be created doesn't make sense.